tests/test_series.py

```diff
@@ -3069,3 +3070,103 @@ def first_arg_not_series(argument_1: int, ser: pd.Series) -> pd.Series:
             ),
             1,
         )
+
+
+def test_diff() -> None:
+    s = pd.Series([1, 1, 2, 3, 5, 8])
+    # int -> float
+    check(assert_type(s.diff(), "pd.Series[float]"), pd.Series, float)
+    # unint -> float
+    check(assert_type(s.astype(np.uint32).diff(), "pd.Series[float]"), pd.Series, float)
+    # float -> float
+    check(assert_type(s.astype(float).diff(), "pd.Series[float]"), pd.Series, float)
+    # datetime.date -> timeDelta
+    check(
+        assert_type(
+            pd.Series(
+                [datetime.datetime.now().date(), datetime.datetime.now().date()]
+            ).diff(),
+            "TimedeltaSeries",
+        ),
+        pd.Series,
+        pd.Timedelta,
+        index_to_check_for_type=-1,
+    )
+    # timestamp -> timedelta
+    times = pd.Series([pd.Timestamp(0), pd.Timestamp(1)])
+    check(
+        assert_type(times.diff(), "TimedeltaSeries"),
+        pd.Series,
+        pd.Timedelta,
+        index_to_check_for_type=-1,
+    )
+    # timedelta -> timedelta64
+    check(
+        assert_type(
+            pd.Series([pd.Timedelta(0), pd.Timedelta(1)]).diff(), "TimedeltaSeries"
+        ),
+        pd.Series,
+        pd.Timedelta,
+        index_to_check_for_type=-1,
+    )
+    # period -> object
+    with pytest_warns_bounded(
+        RuntimeWarning, "overflow encountered in scalar multiply"
+    ):
+        check(
+            assert_type(
+                pd.Series(
+                    [pd.Period("2012-1-1", freq="D"), pd.Period("2012-1-2", freq="D")]
+                ).diff(),
+                "pd.Series[type[object]]",
+            ),
+            pd.Series,
+            object,
+        )
+    # bool -> object
+    check(
+        assert_type(
+            pd.Series([True, True, False, False, True]).diff(),
+            "pd.Series[type[object]]",
+        ),
+        pd.Series,
+        object,
+    )
+    # object -> object
+    check(
+        assert_type(s.astype(object).diff(), "pd.Series[type[object]]"),
+        pd.Series,
+        object,
+    )
+    # Baseoffset (object) -> object
+    # from pandas.tseries.frequencies import to_offset
+    # pd.Series([to_offset("5min"), to_offset("1D1h")]).diff()
+    # pd.Series([pd.DateOffset(days=1), pd.DateOffset(days=2)]).diff()
```

Comment:
Not sure why this is there?

Reply:
sorry I left some mess around. I'll clean it